go/vt/wrangler/traffic_switcher.go

```diff
@@ -995,6 +1016,38 @@ func (ts *trafficSwitcher) changeTableSourceWrites(ctx context.Context, access a
 	})
 }
 
+// executeLockTablesOnSource executes a LOCK TABLES tb1 READ, tbl2 READ,... statement on each
+// source shard's primary tablet using a non-pooled connection as the DBA user. The connection
+// is closed when the LOCK TABLES statement returns, so we immediately release the LOCKs.
+func (ts *trafficSwitcher) executeLockTablesOnSource(ctx context.Context) error {
+	ts.Logger().Infof("Locking (and then immediately unlocking) the following tables on source keyspace %v: %v", ts.SourceKeyspaceName(), ts.Tables())
+	if ts.Tables() == nil || len(ts.Tables()) == 0 {
+		return vterrors.Errorf(vtrpc.Code_INTERNAL, "no tables found in the source keyspace %v associated with the %s workflow", ts.SourceKeyspaceName(), ts.WorkflowName())
+	}
+
+	sb := strings.Builder{}
+	sb.WriteString("LOCK TABLES ")
+	for _, tableName := range ts.Tables() {
+		sb.WriteString(fmt.Sprintf("%s READ,", sqlescape.EscapeID(tableName)))
+	}
+	// trim extra trailing comma
+	lockStmt := sb.String()[:sb.Len()-1]
+
+	return ts.ForAllSources(func(source *workflow.MigrationSource) error {
+		primary := source.GetPrimary()
+		if primary == nil {
+			return vterrors.Errorf(vtrpc.Code_INTERNAL, "no primary found for source shard %s", source.GetShard())
+		}
+		tablet := primary.Tablet
+		_, err := ts.wr.ExecuteFetchAsDba(ctx, tablet.Alias, lockStmt, 1, false, true)
```

Comment:
So `ExecuteFetchAsDba` will release the locks automatically since it doesn't use a pool and the connection will be closed after the locking succeeds, right?

Reply:
Right. I know this for sure because initially I _*wanted*_ to hold the locks and saw that they were getting released immediately and the LOCK/UNLOCK statements were done with different mysql connection IDs.

Reply:
Cool, maybe we can add a comment to that effect here.